backend/pkg/factory/kafka/kafka.go

```diff
@@ -273,7 +274,14 @@ func NewKgoConfig(cfg config.Kafka, logger *zap.Logger, metricsNamespace string)
 	}
 
 	if cfg.TLS.Enabled {
-		tlsConfig, err := cfg.TLS.TLSConfig()
+		// Extract hostname from the first broker URL
+		_, host, err := netpkg.ParseHostMaybeScheme(cfg.Brokers[0])
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse broker url scheme: %w", err)
+		}
+		hostname, _ := netpkg.SplitHostPortDefault(host, 9092)
+
+		tlsConfig, err := cfg.TLS.TLSConfigWithReloader(ctx, logger, hostname)
```

Comment:
I think this wouldn't work, because the `ctx` will quit the Go routine that watches the hot-reloading process. I think the context that is passed into this function is the request context though, and therefore rather short-lived.

Reply:
good catch I will review this.